### ISSUE TITLE
About the use of classify in detect.py

### ISSUE THREAD

**Issue by Zengyf-CVer** [ultralytics/yolov5#5329]
@glenn-jocher 
How to use this `classify` in detect.py? How to write the command line?
https://github.com/ultralytics/yolov5/blob/9c31a66f5dfdf5b7f465ef20a470ad7e2a199354/detect.py#L186-L188

**Comment by glenn-jocher** [comment]
@Zengyf-CVer yes some organizations have created two stage (detect-classify) solutions. In general they may be useful for reducing FPs. See TowerScout for a deployed example of YOLOv5 paired with a second stage EfficientNet model.
https://groups.ischool.berkeley.edu/TowerScout/

<img width="1513" alt="Screenshot 2021-06-09 at 15 14 25" src="https://user-images.githubusercontent.com/26833433/121361242-6632ff80-c935-11eb-8e39-3f5e02bc50a0.png">

See also 'Second-stage classifier in detect.py' https://github.com/ultralytics/yolov5/issues/2260

**Comment by Zengyf-CVer** [comment]
@glenn-jocher 
Are all pre-trained models of YOLOv5 currently trained with this classifier?
https://github.com/ultralytics/yolov5/blob/9c31a66f5dfdf5b7f465ef20a470ad7e2a199354/models/common.py#L459-L469
Can I use your pre-trained model, then set classify to True, and then use the two-stage classifier?
https://github.com/ultralytics/yolov5/blob/9c31a66f5dfdf5b7f465ef20a470ad7e2a199354/detect.py#L77

**Comment by glenn-jocher** [comment]
@Zengyf-CVer `Classify()` head is used for training experimental YOLOv5 classifier models using classifier branch:
https://github.com/ultralytics/yolov5/blob/classifier/classifier.py

We do not have any classification models available, you have to create your own classifier that aligns with the detector classes, i.e. train a classifier on COCO box crops (classes 0-79) to use here, or similar for your custom dataset.

**Comment by Zengyf-CVer** [comment]
@glenn-jocher 
I understand, thank you very much.

**Comment by junzz827** [comment]
Hi Glen san.
I would like to take this opportunity to ask a question.
I am also trying second-stage classification.
When a two-class classification model that classifies cat patterns (stripes or single colors) trained by ResNet18 is loaded into yolov5, the result [0,16] is stored in pred_cls1 of apply_classifier in general.py, but pred_cls2 is [ It will be rewritten to 0,0]. ([0,16] is [person, cat])
Maybe I need to have a dataset with n = 92, which is the original 90 dataset plus my cat pattern dataset?

**Comment by glenn-jocher** [comment]
@junzz827 the basic example we have implemented in the code is a confirm/reject classifier which is trained on the exact same classes as the detection model. i.e. for COCO you would need an 80 classes model with the COCO classes:
https://github.com/ultralytics/yolov5/blob/b8f979bafab6db020d86779b4b40619cd4d77d57/data/coco.yaml#L16-L27

Of course there are many other ways to implement, such as subcategorization, i.e. if person detected then classify as man or woman, etc.

**Comment by junzz827** [comment]
Thank you for your reply.
The subcategorization you were talking about is exactly what I want to implement.
Next, I try to pass the cropped image to the Torch model with normal detection.
I would like to do the following only when an image of a cat is detected, is this idea okay?
>pred2 = modelc(torch.Tensor(imc).to(d.device)).argmax(1)

**Comment by glenn-jocher** [comment]
@junzz827 yes, the idea is to pass the crop of the detection to the classifier to verify/subcategorize it, i.e. get a second opinion from an independent model.

**Comment by Dr395** [comment]
@junzz827 Hello friend, is your problem about yolo subcategorization solved? Can you share a solution? Thank you！！

**Comment by TheoooE** [comment]
@junzz827 Hello, I also encountered the problem of subcategorization. I would appreciate it if you could share!

**Comment by junzz827** [comment]
@Dr395 @TheoooE 
I'm sorry I didn't notice your inquiry!
My solution took advantage of the'save_crop' that came with detect.py.
When a specific target is found (for example, when person = 0), the cropped image of person is sent as'imcrop' to an external secondclassify.py. 
result = secondclassify (imcrop, targetNo.)
secondclassify.py uses the training model of Resnet18 and returns the classification result of the input image.
We cannot publish it due to the client's circumstances, but we will answer your questions as much as possible.

**Comment by TheoooE** [comment]
@junzz827

**Comment by TheoooE** [comment]
@junzz827 
Thanks for your reply. 
This gives me inspiration, and I will try to advance according to your train of thought.

**Comment by anniexxsun** [comment]
@junzz827 Hi there!! Thank you for your reply, it is helpful for a project that I am currently trying to work on as well! I am wondering if you were able to evaluate the results of the two stage model (yolo+resnet) from end to end? from my understanding, the second stage classifier model would need to be trained separately from the yolo, since the detect.py file just takes a pretrained weight. 

Would the validation and test results from the second stage model alone deem suitable as the performance of the overall combination of yolo and resnet? Thank you! Hope the question makes sense!
